### DISCUSSION TITLE
Triple Exponential Moving Average (TEMA)

### DISCUSSION THREAD

**Discussion by DaveSkender** [DaveSkender/Stock.Indicators#808] [Indicators]
Community discussion about [Triple Exponential Moving Average](https://dotnet.stockindicators.dev/indicators/Tema/#content)

**Comment by LibertyBot2019** [comment]
Hi, DaveSkender. Thank you for the indicators. Recently I tried your Triple Exponential Moving Average(TEMA). It seems the output is not correct.  At least the Tema has null values for the first 3*lookBackPeriod - 3 records. But your TEMA output only has a lookBackPeriod - 1 null record. Can you double check this TEMA indicator?

**Reply by DaveSkender** [reply]
See #770 for an explanation.

We use a more efficient single-pass approach.  It’s the same original formula, but it will produce usable*** values earlier and processes 2x faster.

*** convergence to optimal precision is still after `3*N + 100` periods.

**Reply by DaveSkender** [reply]
Adding documentation updates to clarify

- #1270
